### PR TITLE
Ensure relocatable config files

### PR DESCRIPTION
**Pull request by Pro**
Using CMAKE_INSTALL_PREFIX causes build issues when you relocate the installed CMake package, i.e., as it is done in conan.

CMAKE_INSTALL_PREFIX should not be used in the cmake.in files directly. The `@PACKAGE_INIT@` macro initializes `PACKAGE_PREFIX_DIR` correspondingly.

See also:
https://github.com/ignitionrobotics/ign-tools/pull/30
https://github.com/osrf/sdformat/pull/419
https://github.com/osrf/gazebo/issues/2782

cc @stefanbuettner

### CODE REVIEW

File: cmake/pkgconfig/ignition.pc.in [hunk 1]
```diff
@@ -1,4 +1,4 @@
-prefix=@CMAKE_INSTALL_PREFIX@
+prefix=${pcfiledir}/../..
```
Comment (scpeters) [inline cmake/pkgconfig/ignition.pc.in:1]: as discussed in dartsim/dart#1527 (comment), this relative path is incorrect on Ubuntu

**Review by scpeters — CHANGES_REQUESTED** [review]

File: cmake/pkgconfig/ignition.pc.in (hunk 1 above)
Comment (scpeters) [reply cmake/pkgconfig/ignition.pc.in]: I've updated the `.pc.in` templates for downstream ignition packages in 1e39fa0 and the `.pc` file for ign-cmake2 itself in b95d20f

**Comment by scpeters** [comment]
I think it's good now with my fixes, based on the approach in https://github.com/dartsim/dart/pull/1529 and https://github.com/osrf/sdformat/pull/419, but I'd like another set of eyes and some testing if possible

cc @j-rivero @caguero

**Review by scpeters — APPROVED** [review]

**Comment by scpeters** [comment]
perhaps we can test this by including it in a prerelease (see #147)

**Comment by scpeters** [comment]
I'm going to merge this now and include in a prerelease to get further testing

**Comment by j-rivero** [comment]
> I'm going to merge this now and include in a prerelease to get further testing

You were a bit faster merging than I commenting: I tried this today, seems to work fine for all the scenarios I tried: colcon workspace to build ign-math7, simulate a release in Debian packaging system (using `-DCMAKE_INSTALL_LIBDIR=lib/aarch64-linux-gnu`, would be great to set a test for this) or just manually inspecting the file with the default no-arguments. I see no reason not to merge this one. Nice work.

**Comment by scpeters** [comment]
> > I'm going to merge this now and include in a prerelease to get further testing
> 
> You were a bit faster merging than I was commenting: I tried this today, seems to work fine for all the scenarios I tried: colcon workspace to build ign-math7, simulate a release in Debian packaging system (using `-DCMAKE_INSTALL_LIBDIR=lib/aarch64-linux-gnu`, would be great to set a test for this) or just manually inspecting the file with the default no-arguments. I see no reason not to merge this one. Nice work.

sorry for not waiting! we can see how the prerelease goes

**Pull request merged**